elide-standalone/src/test/java/com/yahoo/elide/standalone/hooks/CreationLogHook.java

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright 2019, Yahoo Inc.
+ * Licensed under the Apache License, Version 2.0
+ * See LICENSE file in project root for terms.
+ */
+
+package com.yahoo.elide.standalone.hooks;
+
+import com.yahoo.elide.annotation.Hook;
+import com.yahoo.elide.annotation.OnCreatePreCommit;
+import com.yahoo.elide.functions.LifeCycleHook;
+import com.yahoo.elide.security.ChangeSpec;
+import com.yahoo.elide.security.RequestScope;
+import com.yahoo.elide.standalone.models.Post;
+import lombok.extern.slf4j.Slf4j;
+
+import java.util.Optional;
+
+@Hook(lifeCycle = OnCreatePreCommit.class)
+@Slf4j
+public class CreationLogHook implements LifeCycleHook<Post> {
```

Comment:
Can `LifeCycleHook` be implemented with an interface or abstract class template and elide would scan and bind the hook to all the concrete entities?

Reply:
The short answer is no.  

Elide attempts to bind an entity (if not already bound) when registering a hook.   We wouldn't want to bind (or attempt to bind) every subclass of an abstract class or every implementation of an interface.   Binding down the inheritance tree could produce unintended consequences.

Life cycle events as implemented are bound to specific classes.   When an event is fired, it is fired for only one class (tied to the persistent resource), not to its interfaces, subclasses, etc.   My point here is that there are no global topics - only class specific ones.  This means we can't avoid the problem above by binding an interface (instead of all implementations).

I believe what you are getting at is to provide some kind of way to apply a hook to many different classes at once. 

There are two solutions:
1. For global logging, Elide has another annotation - `@Audit` - which can apply broad logging policies across a set of models.
2. There are three ways to register hooks in Elide:
  1. By putting annotations on the Elide model itself (the original approach)
  2. By calling methods in the Entity Dictionary to bind a function.
  3. This new scanning approach which leverages (2).

It is entirely possible to define an interface (or abstract class) like:
```java
public interface LogOnCreate {
    @OnCreatePreCommit
    default void logCreate() { //Do some logging }
}
```

Any entity that wanted this general provision could implement this interface.